test/sparse/umfpack.jl

```diff
@@ -1,156 +1,169 @@
+    for (Tin, Tout) in (
+            (Complex32, Complex128),
+            (Complex64, Complex128),
+            (Complex128, Complex128),
+            (Float16, Float64),
+            (Float32, Float64),
+            (Float64, Float64),
+            (Int, Float64),
+        )
+
+        F = lufact(sparse(ones(Tin, 1, 1)))
+        L = sparse(ones(Tout, 1, 1))
+        @test F[:p] == F[:q] == [1]
+        @test F[:Rs] == [1.0]
+        @test F[:L] == F[:U] == L
+        @test F[:(:)] == (L, L, [1], [1], [1.0])
+    end
 end
 
-#size(::UmfpackLU)
-let
-    m = n = 1
-    F = lufact(sparse(ones(m, n)))
-    @test size(F) == (m, n)
-    @test size(F, 1) == m
-    @test size(F, 2) == n
-    @test size(F, 3) == 1
-    @test_throws ArgumentError size(F,-1)
+@testset "BigFloat, Complex{BigFloat} throws error in lufact" begin
+    for T in (BigFloat, Complex{BigFloat})
+        @test_throws ArgumentError lufact(sparse(ones(T, 1, 1)))
+    end
+end
+
+@testset "size(::UmfpackLU)" begin
+    let
+        m = n = 1
+        F = lufact(sparse(ones(m, n)))
+        @test size(F) == (m, n)
+        @test size(F, 1) == m
+        @test size(F, 2) == n
+        @test size(F, 3) == 1
+        @test_throws ArgumentError size(F,-1)
+    end
 end
 
-let
-    a = rand(5)
-    @test_throws ArgumentError Base.SparseArrays.UMFPACK.solve!(a, lufact(speye(5,5)), a, Base.SparseArrays.UMFPACK.UMFPACK_A)
-    aa = complex(a)
-    @test_throws ArgumentError Base.SparseArrays.UMFPACK.solve!(aa, lufact(complex(speye(5,5))), aa, Base.SparseArrays.UMFPACK.UMFPACK_A)
+@testset "aliased input and output array throws ArgumentError" begin
+    let
+        a = rand(5)
+        @test_throws ArgumentError Base.SparseArrays.UMFPACK.solve!(a, lufact(speye(5,5)), a, Base.SparseArrays.UMFPACK.UMFPACK_A)
+        aa = complex(a)
+        @test_throws ArgumentError Base.SparseArrays.UMFPACK.solve!(aa, lufact(complex(speye(5,5))), aa, Base.SparseArrays.UMFPACK.UMFPACK_A)
+    end
 end
 
-#18246,18244-lufact sparse pivot
-let A = speye(4)
-    A[1:2,1:2] = [-.01 -200; 200 .001]
-    F = lufact(A)
-    @test F[:p] == [3 ; 4 ; 2 ; 1]
+@testset "issues #18246, #18244-lufact sparse pivot" begin
+    let A = speye(4)
+        A[1:2,1:2] = [-.01 -200; 200 .001]
+        F = lufact(A)
+        @test F[:p] == [3 ; 4 ; 2 ; 1]
+    end
 end
 
-# Test that A[c|t]_ldiv_B!{T<:Complex}(X::StridedMatrix{T}, lu::UmfpackLU{Float64},
-# B::StridedMatrix{T}) works as expected.
-let N = 10, p = 0.5
-    A = N*speye(N) + sprand(N, N, p)
-    X = zeros(Complex{Float64}, N, N)
-    B = complex.(rand(N, N), rand(N, N))
-    luA, lufA = lufact(A), lufact(Array(A))
-    @test A_ldiv_B!(copy(X), luA, B) ≈ A_ldiv_B!(copy(X), lufA, B)
-    @test At_ldiv_B!(copy(X), luA, B) ≈ At_ldiv_B!(copy(X), lufA, B)
-    @test Ac_ldiv_B!(copy(X), luA, B) ≈ Ac_ldiv_B!(copy(X), lufA, B)
+@testset "# Test that A[c|t]_ldiv_B!{T<:Complex}(X::StridedMatrix{T}, lu::UmfpackLU{Float64}, # B::StridedMatrix{T}) works as expected." begin
```

Comment:
I also do not see a second one – there are some testset strings with "issue #123" in them, but that's fine.